File: 최경열/DFS/house.py
n = int(input())
graph = []
arr = []
count = 0
result = 0
for i in range(n):
    graph.append(list(map(int, input())))


def dfs(x, y):
    global count
    if x <= -1 or x >= n or y <= -1 or y >= n:
        return False
    if graph[x][y] == 1:
        graph[x][y] = 0
        count += 1
        dfs(x-1, y)
        dfs(x, y-1)
        dfs(x+1, y)
        dfs(x, y+1)
        return True
    return False


for i in range(n):
    for j in range(n):
        if dfs(i, j) == True:
            result += 1
            arr.append(count)
            count = 0
arr.sort()
print(result)
for i in arr:
    print(i)

File: 최경열/DFS/단지번호붙이기/house.js
const readline =require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
let input=[]
rl.on("line",function(line){
    input.push(line)
}).on("close",function(){
    const n = Number(input[0])
    const graph=[]
    const arr=[]
    let count =0
    let result=0
    for(let i=1; i<=n; i++){
        graph.push(input[i].split("").map(Number));
    }
    function dfs(x,y){
        if(x <= -1 || x >= n || y <= -1 || y >= n) return false;
        if(graph[x][y]==1){
            graph[x][y]=0
            count+=1
            dfs(x-1,y)
            dfs(x,y-1)
            dfs(x+1,y)
            dfs(x,y+1)
            return true
        }
        return false
    }

    for(let i =0; i<n; i++){
        for(let j=0; j<n; j++){
            if(dfs(i,j)==true){
                result+=1
                arr.push(count)
                count=0
            }
        }
    }
    
    arr.sort((a,b)=>a-b)
    console.log(result)

    for(let i in arr){
        console.log(arr[i])
    }
})